### ISSUE TITLE
CVD Dataset Analysis

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/ML-Crate#649]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : CVD Dataset Analysis
:red_circle: **Aim** : The aim is to analyze the dataset using machine learning methods.
:red_circle: **Dataset** : https://www.kaggle.com/datasets/sanazkhanjani/heart-diseases-dataset
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by arijitde92** [comment]
Full name : Arijit De

GitHub Profile Link : [https://github.com/arijitde92](https://github.com/arijitde92)

Participant ID (If not, then put NA) : NA

Approach for this Project : 

1. Exploratory Data Analysis (EDA) to clean the data, handle missing values, and visualize patterns.
2. Feature Selection to only use essential features and remove the non-essential features by analyzing the correlation between target class and different features.
3. Feature Engineering using PCA.
4. Dataset Division - Divide the dataset into train, validation and testing sets.
5. Model Experiments: I will implement and compare models using ML algorithms like Random Forest, MLP. Also use ensemble models like XGBoost and AdaBoost. Each model will be trained on the dataset and evaluated using accuracy, precision, recall, and F1 score. The best model will be identified based on these metrics.
6. Finetuning the hyperparameters to try to optimize the performance.

What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)
VSOC'24

Hi @abhisheks008 , I would like to work on this issue.

**Comment by abhisheks008** [comment]
Implement 6-7 models for this project.
Assigned @arijitde92

**Comment by arijitde92** [comment]
> Implement 6-7 models for this project.
> Assigned @arijitde92 

Thanks
